### PR TITLE
fix: pass in actor to api.can call

### PR DESCRIPTION
**Pull request by barnabasJ**
Addresses #842

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [ ] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by barnabasJ** [comment]
Still need to handle the `{:ok, false}`  and `{:ok, false, exception}` case in some way

**Comment by zachdaniel** [comment]
Ah, right. I think adding the error to the main query should do it. I believe there is an option to tell it to return the exception so that we never get `{:ok, false}`, it will always have the exception.

**Comment by barnabasJ** [comment]
The test in my project is still failing, I'm doing a query without authorization. I think if `authorize?` is `false` we maybe need to skip the call to `Api.can`?

**Comment by zachdaniel** [comment]
Yep, you're totally right.

File: lib/ash/actions/read/relationships.ex [hunk 1]
```diff
@@ -165,33 +165,50 @@ defmodule Ash.Actions.Read.Relationships do
...
-            {:ok, true, authorized_through_query} ->
-              {:ok,
-               [
-                 {clear_lateral_join_source(source_query), relationship.source_attribute,
-                  relationship.source_attribute_on_join_resource, relationship},
-                 {clear_lateral_join_source(authorized_through_query),
-                  relationship.destination_attribute_on_join_resource,
-                  relationship.destination_attribute, join_relationship}
-               ]}
+          if source_query.context[:private][:authorize?] do
```
Comment (barnabasJ) [inline lib/ash/actions/read/relationships.ex:168]: It feels like I'm adding a lot of `if` statements recently, but I was unable to come up with something better in the short term.

Do you have some rule of thumb for things like this? Would you refactor some things out into smaller functions to circumvent this or is there another way?

Comment (zachdaniel) [reply lib/ash/actions/read/relationships.ex:168]: Yeah, if there is a need to refactor I'd focus on extracting something out into a named function. Often I don't end up doing this, much to the chagrin of others. Feel free to do so if you like :)

Comment (barnabasJ) [reply lib/ash/actions/read/relationships.ex:168]: I'm ok with merging it like this for now, I still have a couple other tests that fail with the new read action code. That has priority for me right now.

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**